app/scripts/lib/nonce-tracker.js

```diff
@@ -28,18 +29,26 @@ class NonceTracker {
     // calculate next nonce
     // we need to make sure our base count
     // and pending count are from the same block
+    const localNonceHex = this._getLocalNonce(address)
+    let localNonce = parseInt(localNonceHex, 16)
+    try {
+      assert(Number.isInteger(localNonce), `nonce-tracker - localNonce is not an integer - got: (${typeof localNonce}) "${localNonce}"`)
```

Comment:
seems silly to throw an error via assert and then just catch it and throw out the error
if statement would be more correct